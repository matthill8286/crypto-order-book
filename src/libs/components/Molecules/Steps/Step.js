function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { Typo } from "../../Atoms/Typography";
import { media } from "../../../styles/media";
import { styled } from "../../../styles/styled";
import { StepIcon } from "./StepIcon";
var StepRoot = styled.div.withConfig({
  displayName: "Step__StepRoot",
  componentId: "sc-18g6d9p-0"
})(["cursor:", ";flex:1;position:relative;padding:0 ", "px;"], function (_ref) {
  var pointer = _ref.pointer;
  return pointer ? 'pointer' : 'default';
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.defaultSpacing;
});
var StepLabelRoot = styled.span.withConfig({
  displayName: "Step__StepLabelRoot",
  componentId: "sc-18g6d9p-1"
})(["flex-direction:column;display:flex;align-items:center;", " ", ""], function (_ref3) {
  var elementPosition = _ref3.elementPosition;
  return elementPosition === 'first' && 'align-items: flex-start;';
}, function (_ref4) {
  var elementPosition = _ref4.elementPosition;
  return elementPosition === 'last' && 'align-items: flex-end;';
});
var StepDescription = styled(Typo).withConfig({
  displayName: "Step__StepDescription",
  componentId: "sc-18g6d9p-2"
})(["", "{display:none;}"], media.maxMd);
export var Step = function Step(props) {
  var _props$color = props.color,
      color = _props$color === void 0 ? 'primary' : _props$color,
      _props$elementPositio = props.elementPosition,
      elementPosition = _props$elementPositio === void 0 ? 'default' : _props$elementPositio,
      _props$active = props.active,
      active = _props$active === void 0 ? false : _props$active,
      _props$completed = props.completed,
      completed = _props$completed === void 0 ? false : _props$completed,
      _props$stepIndex = props.stepIndex,
      stepIndex = _props$stepIndex === void 0 ? 0 : _props$stepIndex,
      _props$title = props.title,
      title = _props$title === void 0 ? '' : _props$title,
      _props$description = props.description,
      description = _props$description === void 0 ? '' : _props$description,
      icon = props.icon,
      connector = props.connector,
      _onClick = props.onClick,
      _props$variant = props.variant,
      variant = _props$variant === void 0 ? 'prominent' : _props$variant,
      other = _objectWithoutProperties(props, ["color", "elementPosition", "active", "completed", "stepIndex", "title", "description", "icon", "connector", "onClick", "variant"]);

  var stepIconProps = {
    active: active,
    completed: completed,
    color: color,
    variant: variant,
    customIcon: icon
  };
  var textColor = active || completed ? color : 'grey2';
  return React.createElement(StepRoot, _extends({}, other, {
    pointer: !!_onClick,
    onClick: function onClick() {
      return _onClick && _onClick(stepIndex);
    }
  }), connector, variant != 'uniform' ? React.createElement(StepLabelRoot, {
    elementPosition: elementPosition
  }, title && React.createElement(Typo, {
    tag: "span",
    color: textColor,
    weight: active || completed ? 'semibold' : 'regular'
  }, title), description && React.createElement(StepDescription, {
    tag: "span",
    color: textColor
  }, description), React.createElement(StepIcon, stepIconProps)) : React.createElement(StepIcon, stepIconProps));
};
Step.displayName = 'Step';
//# sourceMappingURL=Step.js.map