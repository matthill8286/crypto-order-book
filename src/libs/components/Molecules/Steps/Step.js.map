{"version":3,"sources":["../../../../src/components/Molecules/Steps/Step.tsx"],"names":["React","Typo","media","styled","StepIcon","StepRoot","div","pointer","theme","defaultSpacing","StepLabelRoot","span","elementPosition","StepDescription","maxMd","Step","props","color","active","completed","stepIndex","title","description","icon","connector","onClick","variant","other","stepIconProps","customIcon","textColor","displayName"],"mappings":";;;;;;AAAA,OAAOA,KAAP,MAA0B,OAA1B;AACA,SAASC,IAAT;AACA,SAASC,KAAT;AACA,SAASC,MAAT;AAEA,SAASC,QAAT;AAQA,IAAMC,QAAQ,GAAGF,MAAM,CAACG,GAAV;AAAA;AAAA;AAAA,+DACF;AAAA,MAAGC,OAAH,QAAGA,OAAH;AAAA,SAAkBA,OAAO,GAAG,SAAH,GAAe,SAAxC;AAAA,CADE,EAIC;AAAA,MAAGC,KAAH,SAAGA,KAAH;AAAA,SAAeA,KAAK,CAACC,cAArB;AAAA,CAJD,CAAd;AAOA,IAAMC,aAAa,GAAGP,MAAM,CAACQ,IAAV;AAAA;AAAA;AAAA,wEAKf;AAAA,MAAGC,eAAH,SAAGA,eAAH;AAAA,SAAyBA,eAAe,KAAK,OAApB,IAA+B,0BAAxD;AAAA,CALe,EAMf;AAAA,MAAGA,eAAH,SAAGA,eAAH;AAAA,SAAyBA,eAAe,KAAK,MAApB,IAA8B,wBAAvD;AAAA,CANe,CAAnB;AASA,IAAMC,eAAe,GAAGV,MAAM,CAACF,IAAD,CAAT;AAAA;AAAA;AAAA,4BACjBC,KAAK,CAACY,KADW,CAArB;AAKA,OAAO,IAAMC,IAA4C,GAAG,SAA/CA,IAA+C,CAAAC,KAAK,EAAI;AAAA,qBAc/DA,KAd+D,CAEjEC,KAFiE;AAAA,MAEjEA,KAFiE,6BAEzD,SAFyD;AAAA,8BAc/DD,KAd+D,CAGjEJ,eAHiE;AAAA,MAGjEA,eAHiE,sCAG/C,SAH+C;AAAA,sBAc/DI,KAd+D,CAIjEE,MAJiE;AAAA,MAIjEA,MAJiE,8BAIxD,KAJwD;AAAA,yBAc/DF,KAd+D,CAKjEG,SALiE;AAAA,MAKjEA,SALiE,iCAKrD,KALqD;AAAA,yBAc/DH,KAd+D,CAMjEI,SANiE;AAAA,MAMjEA,SANiE,iCAMrD,CANqD;AAAA,qBAc/DJ,KAd+D,CAOjEK,KAPiE;AAAA,MAOjEA,KAPiE,6BAOzD,EAPyD;AAAA,2BAc/DL,KAd+D,CAQjEM,WARiE;AAAA,MAQjEA,WARiE,mCAQnD,EARmD;AAAA,MASjEC,IATiE,GAc/DP,KAd+D,CASjEO,IATiE;AAAA,MAUjEC,SAViE,GAc/DR,KAd+D,CAUjEQ,SAViE;AAAA,MAWjEC,QAXiE,GAc/DT,KAd+D,CAWjES,OAXiE;AAAA,uBAc/DT,KAd+D,CAYjEU,OAZiE;AAAA,MAYjEA,OAZiE,+BAYvD,WAZuD;AAAA,MAa9DC,KAb8D,4BAc/DX,KAd+D;;AAgBnE,MAAMY,aAA4B,GAAG;AACnCV,IAAAA,MAAM,EAANA,MADmC;AAEnCC,IAAAA,SAAS,EAATA,SAFmC;AAGnCF,IAAAA,KAAK,EAALA,KAHmC;AAInCS,IAAAA,OAAO,EAAPA,OAJmC;AAKnCG,IAAAA,UAAU,EAAEN;AALuB,GAArC;AAQA,MAAMO,SAAsB,GAAGZ,MAAM,IAAIC,SAAV,GAAsBF,KAAtB,GAA8B,OAA7D;AAEA,SACE,oBAAC,QAAD,eAAcU,KAAd;AAAqB,IAAA,OAAO,EAAE,CAAC,CAACF,QAAhC;AAAyC,IAAA,OAAO,EAAE;AAAA,aAAMA,QAAO,IAAIA,QAAO,CAACL,SAAD,CAAxB;AAAA;AAAlD,MACGI,SADH,EAEGE,OAAO,IAAI,SAAX,GACC,oBAAC,aAAD;AAAe,IAAA,eAAe,EAAEd;AAAhC,KACGS,KAAK,IACJ,oBAAC,IAAD;AACE,IAAA,GAAG,EAAC,MADN;AAEE,IAAA,KAAK,EAAES,SAFT;AAGE,IAAA,MAAM,EAAEZ,MAAM,IAAIC,SAAV,GAAsB,UAAtB,GAAmC;AAH7C,KAIGE,KAJH,CAFJ,EASGC,WAAW,IACV,oBAAC,eAAD;AAAiB,IAAA,GAAG,EAAC,MAArB;AAA4B,IAAA,KAAK,EAAEQ;AAAnC,KACGR,WADH,CAVJ,EAcE,oBAAC,QAAD,EAAcM,aAAd,CAdF,CADD,GAkBC,oBAAC,QAAD,EAAcA,aAAd,CApBJ,CADF;AAyBD,CAnDM;AAqDPb,IAAI,CAACgB,WAAL,GAAmB,MAAnB","sourcesContent":["import React, { FC } from 'react'\nimport { Typo } from '@/components/Atoms/Typography'\nimport { media } from '@/styles/media'\nimport { styled } from '@/styles/styled'\nimport { ThemeColors } from '@/types/theme'\nimport { StepIcon } from './StepIcon'\nimport {\n  ElementPosition,\n  StepIconProps,\n  StepPrivateProps,\n  StepPublicProps,\n} from './Steps.interface'\n\nconst StepRoot = styled.div<{ pointer: boolean }>`\n  cursor: ${({ pointer }) => (pointer ? 'pointer' : 'default')};\n  flex: 1;\n  position: relative;\n  padding: 0 ${({ theme }) => theme.defaultSpacing}px;\n`\n\nconst StepLabelRoot = styled.span<{ elementPosition: ElementPosition }>`\n  flex-direction: column;\n  display: flex;\n  align-items: center;\n\n  ${({ elementPosition }) => elementPosition === 'first' && 'align-items: flex-start;'}\n  ${({ elementPosition }) => elementPosition === 'last' && 'align-items: flex-end;'}\n`\n\nconst StepDescription = styled(Typo)`\n  ${media.maxMd} {\n    display: none;\n  }\n`\nexport const Step: FC<StepPublicProps & StepPrivateProps> = props => {\n  const {\n    color = 'primary',\n    elementPosition = 'default',\n    active = false,\n    completed = false,\n    stepIndex = 0,\n    title = '',\n    description = '',\n    icon,\n    connector,\n    onClick,\n    variant = 'prominent',\n    ...other\n  } = props\n\n  const stepIconProps: StepIconProps = {\n    active,\n    completed,\n    color,\n    variant,\n    customIcon: icon,\n  }\n\n  const textColor: ThemeColors = active || completed ? color : 'grey2'\n\n  return (\n    <StepRoot {...other} pointer={!!onClick} onClick={() => onClick && onClick(stepIndex)}>\n      {connector}\n      {variant != 'uniform' ? (\n        <StepLabelRoot elementPosition={elementPosition}>\n          {title && (\n            <Typo\n              tag=\"span\"\n              color={textColor}\n              weight={active || completed ? 'semibold' : 'regular'}>\n              {title}\n            </Typo>\n          )}\n          {description && (\n            <StepDescription tag=\"span\" color={textColor}>\n              {description}\n            </StepDescription>\n          )}\n          <StepIcon {...stepIconProps} />\n        </StepLabelRoot>\n      ) : (\n        <StepIcon {...stepIconProps} />\n      )}\n    </StepRoot>\n  )\n}\n\nStep.displayName = 'Step'\n"],"file":"Step.js"}