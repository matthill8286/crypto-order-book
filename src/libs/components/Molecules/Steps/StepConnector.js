function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { css, styled } from "../../../styles/styled";

var getStepConnectorPositions = function getStepConnectorPositions(_ref) {
  var theme = _ref.theme,
      variant = _ref.variant,
      elementPosition = _ref.elementPosition,
      rightMargin = _ref.rightMargin,
      leftMargin = _ref.leftMargin;
  var positions = {
    uniform: {
      first: css(["left:50%;right:0;"]),
      second: css(["left:-50%;right:50%;"]),
      last: css(["left:-50%;right:50%;"]),
      default: css(["left:-50%;right:50%;"])
    },
    prominent: {
      first: css(["left:10px;right:calc(100% - ", ");"], theme.spacing.base.md),
      second: css(["left:calc(-100% + ", ");right:50%;"], theme.spacing.base.md),
      last: css(["left:-50%;right:10px;"]),
      default: css(["left:-50%;right:50%;"])
    },
    simple: {
      first: css(["display:none;"]),
      second: css(["left:calc(-100% + ", "px);right:calc(50% + ", "px);"], 42 + leftMargin, 28 + rightMargin),
      last: css(["left:calc(-50% + ", "px);right:calc(0% + ", "px);"], 28 + leftMargin, 42 + rightMargin),
      default: css(["left:calc(-50% + ", "px);right:calc(50% + ", "px);"], 28 + leftMargin, 28 + rightMargin)
    }
  };
  return positions[variant][elementPosition];
};

var BottomPositions = {
  uniform: {
    completed: '16px',
    default: '8px'
  },
  prominent: {
    completed: '16px',
    default: '8px'
  },
  simple: {
    completed: '5px',
    default: '5px'
  }
};
export var StepConnectorRoot = styled.div.withConfig({
  displayName: "StepConnector__StepConnectorRoot",
  componentId: "v4n1if-0"
})(function (_ref2) {
  var theme = _ref2.theme,
      _ref2$elementPosition = _ref2.elementPosition,
      elementPosition = _ref2$elementPosition === void 0 ? 'default' : _ref2$elementPosition,
      active = _ref2.active,
      completed = _ref2.completed,
      prevActive = _ref2.prevActive,
      _ref2$variant = _ref2.variant,
      variant = _ref2$variant === void 0 ? 'prominent' : _ref2$variant;
  var leftMargin = active ? 4 : 0;
  var rightMargin = prevActive ? 4 : 0;
  var currentPosition = getStepConnectorPositions({
    theme: theme,
    variant: variant,
    elementPosition: elementPosition,
    leftMargin: leftMargin,
    rightMargin: rightMargin
  });
  var bottomPosition = completed ? BottomPositions[variant]['completed'] : BottomPositions[variant]['default'];
  return css(["position:absolute;bottom:", ";", ";"], bottomPosition, currentPosition);
});
export var StepConnectorLine = styled.span.withConfig({
  displayName: "StepConnector__StepConnectorLine",
  componentId: "v4n1if-1"
})(["border-color:", ";border-style:solid;border-top-width:", ";border-bottom-width:", ";border-radius:", ";display:block;"], function (_ref3) {
  var theme = _ref3.theme,
      _ref3$color = _ref3.color,
      color = _ref3$color === void 0 ? 'primary' : _ref3$color,
      active = _ref3.active,
      completed = _ref3.completed,
      _ref3$elementPosition = _ref3.elementPosition,
      elementPosition = _ref3$elementPosition === void 0 ? 'default' : _ref3$elementPosition;
  return completed || active || elementPosition === 'first' ? theme.color[color] : theme.color.grey2;
}, function (_ref4) {
  var borderWidth = _ref4.borderWidth;
  return borderWidth;
}, function (_ref5) {
  var borderWidth = _ref5.borderWidth;
  return borderWidth;
}, function (_ref6) {
  var theme = _ref6.theme;
  return theme.dimension.borderRadius1;
});
export var StepConnector = function StepConnector(props) {
  return React.createElement(StepConnectorRoot, props, React.createElement(StepConnectorLine, _extends({}, props, {
    borderWidth: props.variant === 'simple' ? '2px' : '4px'
  })));
};
StepConnector.displayName = 'StepConnector';
//# sourceMappingURL=StepConnector.js.map