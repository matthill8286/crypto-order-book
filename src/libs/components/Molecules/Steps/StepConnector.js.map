{"version":3,"sources":["../../../../src/components/Molecules/Steps/StepConnector.tsx"],"names":["React","css","styled","getStepConnectorPositions","theme","variant","elementPosition","rightMargin","leftMargin","positions","uniform","first","second","last","default","prominent","spacing","base","md","simple","BottomPositions","completed","StepConnectorRoot","div","active","prevActive","currentPosition","bottomPosition","StepConnectorLine","span","color","grey2","borderWidth","dimension","borderRadius1","StepConnector","props","displayName"],"mappings":";;AAAA,OAAOA,KAAP,MAA0B,OAA1B;AACA,SAASC,GAAT,EAAcC,MAAd;;AAOA,IAAMC,yBAAyB,GAAG,SAA5BA,yBAA4B,OAMJ;AAAA,MAL5BC,KAK4B,QAL5BA,KAK4B;AAAA,MAJ5BC,OAI4B,QAJ5BA,OAI4B;AAAA,MAH5BC,eAG4B,QAH5BA,eAG4B;AAAA,MAF5BC,WAE4B,QAF5BA,WAE4B;AAAA,MAD5BC,UAC4B,QAD5BA,UAC4B;AAC5B,MAAMC,SAAiC,GAAG;AACxCC,IAAAA,OAAO,EAAE;AACPC,MAAAA,KAAK,EAAEV,GAAF,uBADE;AAKPW,MAAAA,MAAM,EAAEX,GAAF,0BALC;AASPY,MAAAA,IAAI,EAAEZ,GAAF,0BATG;AAaPa,MAAAA,OAAO,EAAEb,GAAF;AAbA,KAD+B;AAmBxCc,IAAAA,SAAS,EAAE;AACTJ,MAAAA,KAAK,EAAEV,GAAF,yCAEkBG,KAAK,CAACY,OAAN,CAAcC,IAAd,CAAmBC,EAFrC,CADI;AAKTN,MAAAA,MAAM,EAAEX,GAAF,yCACiBG,KAAK,CAACY,OAAN,CAAcC,IAAd,CAAmBC,EADpC,CALG;AASTL,MAAAA,IAAI,EAAEZ,GAAF,2BATK;AAaTa,MAAAA,OAAO,EAAEb,GAAF;AAbE,KAnB6B;AAqCxCkB,IAAAA,MAAM,EAAE;AACNR,MAAAA,KAAK,EAAEV,GAAF,mBADC;AAINW,MAAAA,MAAM,EAAEX,GAAF,0DACiB,KAAKO,UADtB,EAEgB,KAAKD,WAFrB,CAJA;AAQNM,MAAAA,IAAI,EAAEZ,GAAF,wDACkB,KAAKO,UADvB,EAEiB,KAAKD,WAFtB,CARE;AAYNO,MAAAA,OAAO,EAAEb,GAAF,yDACe,KAAKO,UADpB,EAEe,KAAKD,WAFpB;AAZD;AArCgC,GAA1C;AAwDA,SAAOE,SAAS,CAACJ,OAAD,CAAT,CAAmBC,eAAnB,CAAP;AACD,CAhED;;AAkEA,IAAMc,eAAe,GAAG;AACtBV,EAAAA,OAAO,EAAE;AACPW,IAAAA,SAAS,EAAE,MADJ;AAEPP,IAAAA,OAAO,EAAE;AAFF,GADa;AAKtBC,EAAAA,SAAS,EAAE;AACTM,IAAAA,SAAS,EAAE,MADF;AAETP,IAAAA,OAAO,EAAE;AAFA,GALW;AAStBK,EAAAA,MAAM,EAAE;AACNE,IAAAA,SAAS,EAAE,KADL;AAENP,IAAAA,OAAO,EAAE;AAFH;AATc,CAAxB;AAeA,OAAO,IAAMQ,iBAAiB,GAAGpB,MAAM,CAACqB,GAAV;AAAA;AAAA;AAAA,GAC5B,iBAOM;AAAA,MANJnB,KAMI,SANJA,KAMI;AAAA,oCALJE,eAKI;AAAA,MALJA,eAKI,sCALc,SAKd;AAAA,MAJJkB,MAII,SAJJA,MAII;AAAA,MAHJH,SAGI,SAHJA,SAGI;AAAA,MAFJI,UAEI,SAFJA,UAEI;AAAA,4BADJpB,OACI;AAAA,MADJA,OACI,8BADM,WACN;AACJ,MAAMG,UAAU,GAAGgB,MAAM,GAAG,CAAH,GAAO,CAAhC;AACA,MAAMjB,WAAW,GAAGkB,UAAU,GAAG,CAAH,GAAO,CAArC;AAEA,MAAMC,eAAe,GAAGvB,yBAAyB,CAAC;AAChDC,IAAAA,KAAK,EAALA,KADgD;AAEhDC,IAAAA,OAAO,EAAPA,OAFgD;AAGhDC,IAAAA,eAAe,EAAfA,eAHgD;AAIhDE,IAAAA,UAAU,EAAVA,UAJgD;AAKhDD,IAAAA,WAAW,EAAXA;AALgD,GAAD,CAAjD;AAQA,MAAMoB,cAAc,GAAGN,SAAS,GAC5BD,eAAe,CAACf,OAAD,CAAf,CAAyB,WAAzB,CAD4B,GAE5Be,eAAe,CAACf,OAAD,CAAf,CAAyB,SAAzB,CAFJ;AAIA,SAAOJ,GAAP,0CAEY0B,cAFZ,EAGID,eAHJ;AAKD,CA7B2B,CAAvB;AAgCP,OAAO,IAAME,iBAAiB,GAAG1B,MAAM,CAAC2B,IAAV;AAAA;AAAA;AAAA,8HACZ;AAAA,MAAGzB,KAAH,SAAGA,KAAH;AAAA,0BAAU0B,KAAV;AAAA,MAAUA,KAAV,4BAAkB,SAAlB;AAAA,MAA6BN,MAA7B,SAA6BA,MAA7B;AAAA,MAAqCH,SAArC,SAAqCA,SAArC;AAAA,oCAAgDf,eAAhD;AAAA,MAAgDA,eAAhD,sCAAkE,SAAlE;AAAA,SACde,SAAS,IAAIG,MAAb,IAAuBlB,eAAe,KAAK,OAA3C,GAAqDF,KAAK,CAAC0B,KAAN,CAAYA,KAAZ,CAArD,GAA0E1B,KAAK,CAAC0B,KAAN,CAAYC,KADxE;AAAA,CADY,EAIR;AAAA,MAAGC,WAAH,SAAGA,WAAH;AAAA,SAAqBA,WAArB;AAAA,CAJQ,EAKL;AAAA,MAAGA,WAAH,SAAGA,WAAH;AAAA,SAAqBA,WAArB;AAAA,CALK,EAMX;AAAA,MAAG5B,KAAH,SAAGA,KAAH;AAAA,SAAeA,KAAK,CAAC6B,SAAN,CAAgBC,aAA/B;AAAA,CANW,CAAvB;AAUP,OAAO,IAAMC,aAAqC,GAAG,SAAxCA,aAAwC,CAAAC,KAAK;AAAA,SACxD,oBAAC,iBAAD,EAAuBA,KAAvB,EACE,oBAAC,iBAAD,eAAuBA,KAAvB;AAA8B,IAAA,WAAW,EAAEA,KAAK,CAAC/B,OAAN,KAAkB,QAAlB,GAA6B,KAA7B,GAAqC;AAAhF,KADF,CADwD;AAAA,CAAnD;AAMP8B,aAAa,CAACE,WAAd,GAA4B,eAA5B","sourcesContent":["import React, { FC } from 'react'\nimport { css, styled } from '@/styles/styled'\nimport {\n  StepConnectorLineProps,\n  StepConnectorPositions,\n  StepConnectorProps,\n} from './Steps.interface'\n\nconst getStepConnectorPositions = ({\n  theme,\n  variant,\n  elementPosition,\n  rightMargin,\n  leftMargin,\n}): StepConnectorPositions => {\n  const positions: StepConnectorPositions = {\n    uniform: {\n      first: css`\n        left: 50%;\n        right: 0;\n      `,\n      second: css`\n        left: -50%;\n        right: 50%;\n      `,\n      last: css`\n        left: -50%;\n        right: 50%;\n      `,\n      default: css`\n        left: -50%;\n        right: 50%;\n      `,\n    },\n    prominent: {\n      first: css`\n        left: 10px;\n        right: calc(100% - ${theme.spacing.base.md});\n      `,\n      second: css`\n        left: calc(-100% + ${theme.spacing.base.md});\n        right: 50%;\n      `,\n      last: css`\n        left: -50%;\n        right: 10px;\n      `,\n      default: css`\n        left: -50%;\n        right: 50%;\n      `,\n    },\n    simple: {\n      first: css`\n        display: none;\n      `,\n      second: css`\n        left: calc(-100% + ${42 + leftMargin}px);\n        right: calc(50% + ${28 + rightMargin}px);\n      `,\n      last: css`\n        left: calc(-50% + ${28 + leftMargin}px);\n        right: calc(0% + ${42 + rightMargin}px);\n      `,\n      default: css`\n        left: calc(-50% + ${28 + leftMargin}px);\n        right: calc(50% + ${28 + rightMargin}px);\n      `,\n    },\n  }\n\n  return positions[variant][elementPosition]\n}\n\nconst BottomPositions = {\n  uniform: {\n    completed: '16px',\n    default: '8px',\n  },\n  prominent: {\n    completed: '16px',\n    default: '8px',\n  },\n  simple: {\n    completed: '5px',\n    default: '5px',\n  },\n}\n\nexport const StepConnectorRoot = styled.div<StepConnectorProps>(\n  ({\n    theme,\n    elementPosition = 'default',\n    active,\n    completed,\n    prevActive,\n    variant = 'prominent',\n  }) => {\n    const leftMargin = active ? 4 : 0\n    const rightMargin = prevActive ? 4 : 0\n\n    const currentPosition = getStepConnectorPositions({\n      theme,\n      variant,\n      elementPosition,\n      leftMargin,\n      rightMargin,\n    })\n\n    const bottomPosition = completed\n      ? BottomPositions[variant]['completed']\n      : BottomPositions[variant]['default']\n\n    return css`\n      position: absolute;\n      bottom: ${bottomPosition};\n      ${currentPosition};\n    `\n  }\n)\n\nexport const StepConnectorLine = styled.span<StepConnectorLineProps>`\n  border-color: ${({ theme, color = 'primary', active, completed, elementPosition = 'default' }) =>\n    completed || active || elementPosition === 'first' ? theme.color[color] : theme.color.grey2};\n  border-style: solid;\n  border-top-width: ${({ borderWidth }) => borderWidth};\n  border-bottom-width: ${({ borderWidth }) => borderWidth};\n  border-radius: ${({ theme }) => theme.dimension.borderRadius1};\n  display: block;\n`\n\nexport const StepConnector: FC<StepConnectorProps> = props => (\n  <StepConnectorRoot {...props}>\n    <StepConnectorLine {...props} borderWidth={props.variant === 'simple' ? '2px' : '4px'} />\n  </StepConnectorRoot>\n)\n\nStepConnector.displayName = 'StepConnector'\n"],"file":"StepConnector.js"}