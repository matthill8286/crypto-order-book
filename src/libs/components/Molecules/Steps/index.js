export * from "./StepConnector";
export * from "./Step";
export * from "./Steps";
export * from "./Steps.interface";
//# sourceMappingURL=index.js.map