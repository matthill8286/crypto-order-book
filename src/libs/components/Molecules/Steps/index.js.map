{"version":3,"sources":["../../../../src/components/Molecules/Steps/index.ts"],"names":[],"mappings":"AAAA;AACA;AACA;AACA","sourcesContent":["export * from './StepConnector'\nexport * from './Step'\nexport * from './Steps'\nexport * from './Steps.interface'\n"],"file":"index.js"}