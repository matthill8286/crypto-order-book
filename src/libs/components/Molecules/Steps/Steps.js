function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React, { Children, cloneElement, isValidElement } from 'react';
import { Card } from "../../Atoms/Card";
import { FlexBox } from "../../Helper/FlexBox";
import { StepConnector } from "./StepConnector";
export var Steps = function Steps(props) {
  var _props$activeStep = props.activeStep,
      activeStep = _props$activeStep === void 0 ? 0 : _props$activeStep,
      _props$completedStep = props.completedStep,
      completedStep = _props$completedStep === void 0 ? 0 : _props$completedStep,
      _props$elevation = props.elevation,
      elevation = _props$elevation === void 0 ? 0 : _props$elevation,
      connector = props.connector,
      children = props.children,
      _props$color = props.color,
      color = _props$color === void 0 ? 'primary' : _props$color,
      _props$variant = props.variant,
      variant = _props$variant === void 0 ? 'prominent' : _props$variant,
      onChange = props.onChange,
      other = _objectWithoutProperties(props, ["activeStep", "completedStep", "elevation", "connector", "children", "color", "variant", "onChange"]);

  var childrenArray = Children.toArray(children);
  var steps = childrenArray.map(function (step, stepIndex) {
    if (isValidElement(step)) {
      var active = activeStep === stepIndex;
      var completed = stepIndex <= completedStep;
      var elementPosition = 'default';
      if (stepIndex === 0) elementPosition = 'first';
      if (stepIndex === 1) elementPosition = 'second';
      if (stepIndex + 1 === Children.count(children)) elementPosition = 'last';
      var connectorProps = {
        elementPosition: elementPosition,
        active: stepIndex <= activeStep,
        prevActive: stepIndex - 1 === activeStep,
        completed: completed,
        color: color,
        variant: variant
      };
      var connectorComponent = isValidElement(connector) ? cloneElement(connector, _objectSpread(_objectSpread({}, connectorProps), connector.props)) : React.createElement(StepConnector, connectorProps);

      var stepProps = _objectSpread(_objectSpread({}, step.props), {}, {
        connector: connectorComponent,
        elementPosition: elementPosition,
        stepIndex: stepIndex,
        active: active,
        completed: completed,
        color: color,
        onClick: onChange,
        variant: variant
      });

      return cloneElement(step, stepProps);
    }
  });
  return React.createElement(Card, _extends({
    elevation: elevation,
    noBorder: ['top', 'right', 'bottom', 'left'],
    surface: "clear",
    padding: "md"
  }, other), React.createElement(FlexBox, {
    display: "flex",
    alignItems: "center",
    flexDirection: "row"
  }, steps));
};
Steps.displayName = 'Steps';
//# sourceMappingURL=Steps.js.map