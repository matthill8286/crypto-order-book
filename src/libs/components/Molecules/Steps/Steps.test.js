import React from 'react';
import { renderWithTheme } from "../../../testRenderer";
import { Steps } from "./Steps";
describe('Steps', function () {
  var props = {
    activeStep: 0,
    completedStep: -1
  };
  it('should render', function () {
    var tree = renderWithTheme(React.createElement(Steps, props));
    expect(tree).toMatchSnapshot();
  });
});
//# sourceMappingURL=Steps.test.js.map