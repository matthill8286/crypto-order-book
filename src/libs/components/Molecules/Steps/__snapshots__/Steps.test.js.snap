// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps should render 1`] = `
<div
  className="Cardstyled__StyledCardWrapper-sc-1gxmhl7-6 jhkssp"
>
  <div
    className="Cardstyled__StyledCard-sc-1gxmhl7-1 gPtiqh"
    display="block"
    elevation={0}
  >
    <div
      className="FlexBox__StyledBox-sc-52meld-0 iVkoxK"
      display="flex"
    />
  </div>
</div>
`;
